lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_appbar.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:ui';
+
+import 'package:app/core/domain/common/entities/common.dart';
+import 'package:app/core/domain/event/entities/event.dart';
+import 'package:app/core/presentation/widgets/image_placeholder_widget.dart';
+import 'package:app/core/utils/image_utils.dart';
+import 'package:app/gen/fonts.gen.dart';
+import 'package:app/theme/typo.dart';
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class GuestEventDetailAppBar extends StatelessWidget {
+  const GuestEventDetailAppBar({super.key, 
+    required this.event,
+    required this.isCollapsed,
+  });
+
+  final Event event;
+  final bool isCollapsed;
+  @override
+  Widget build(BuildContext context) {
+    return SliverAppBar(
+      pinned: true,
+      stretch: true,
+      floating: true,
+      leading: const SizedBox.shrink(),
+      collapsedHeight: 60.w,
+      expandedHeight: 188.w,
+      centerTitle: true,
+      title: 
+          AnimatedOpacity(
+            duration: const Duration(milliseconds: 300),
+            opacity: isCollapsed ? 1 : 0,  
+            child: Text(
+              event.title ?? '',
+              style: Typo.extraMedium.copyWith(
+                fontFamily: FontFamily.switzerVariable,
+              ),
+            ),
+          ),
+      flexibleSpace: FlexibleSpaceBar(
+        expandedTitleScale: 1,
+        collapseMode: CollapseMode.pin,
+        background: _EventDetailCover(event: event),
+      ),
+    );
+  }
+}
+
+
+class _EventDetailCover extends StatelessWidget {
+  const _EventDetailCover({required this.event});
+
```

Comment:
Why we have to separate this, can't it be merge to 1 class only ?

Reply:
@roncoderpush 
The event's cover is part of event sliver app bar, I want to separate for easier to look and separate for single concern